src/vm_pages.rs

```diff
@@ -578,6 +615,39 @@ impl Drop for PtePagePool {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+struct PageRangeInfo {
```

Comment:
Thoughts on using the `VmRegionList` I just added to support shared page ranges instead? It supports multiple regions and RAII-based error recovery, which would simplify the implementation.

Reply:
Yes, noticed it too, but haven't really internalized the purpose.